src/main/java/dev/socketmods/socketnukes/registry/SNRegistry.java

```diff
@@ -113,4 +114,20 @@ public static void initialize() {
         EXPLOSIONS.register(modBus);
         ENTITYTYPES.register(modBus);
     }
+
+    public static ExtendedExplosionType getExplosionType(String name) {
+        return Objects.requireNonNull(EXPLOSION_TYPE_REGISTRY.get().getValue(new ResourceLocation(name)));
+    }
+
+    public static ExtendedExplosionType getExplosionType(ResourceLocation name) {
+        return Objects.requireNonNull(EXPLOSION_TYPE_REGISTRY.get().getValue(name));
+    }
+
+    public static <T extends ForgeRegistryEntry<T>> ResourceLocation getName(T type) {
+        return Objects.requireNonNull(type.getRegistryName());
+    }
+
+    public static <T extends ForgeRegistryEntry<T>> ResourceLocation getName(Supplier<T> supplier) {
+        return getName(supplier.get());
+    }
```

Comment:
Probably better to reference `IForgeRegistryEntry` here instead of `ForgeRegistryEntry`, as the interface is the more-API-ish of the two (`ForgeRegistryEntry` is a convenience implementation with benefits).

Reply:
Missed this when I was coping fixes from `dev-actor` across